src/utilities/strategy-helpers/deployment-helper.ts

```diff
@@ -31,19 +35,51 @@ export async function deploy(kubectl: Kubectl, manifestFilePaths: string[], depl
     inputManifestFiles = updateImagePullSecretsInManifestFiles(inputManifestFiles, TaskInputParameters.imagePullSecrets);
 
     // deployment
-    const deployedManifestFiles = deployManifests(inputManifestFiles, kubectl, isCanaryDeploymentStrategy(deploymentStrategy));
+    const deployedManifestFiles = deployManifests(inputManifestFiles, kubectl, isCanaryDeploymentStrategy(deploymentStrategy), isBlueGreenDeploymentStrategy(deploymentStrategy));
 
     // check manifest stability
     const resourceTypes: Resource[] = KubernetesObjectUtility.getResources(deployedManifestFiles, models.deploymentTypes.concat([KubernetesConstants.DiscoveryAndLoadBalancerResource.service]));
     await checkManifestStability(kubectl, resourceTypes);
 
+    // route blue-green deployments
+    if (blueGreenDeploymentHelper.isBlueGreenDeploymentStrategy()) {
+        await routeBlueGreen(kubectl, inputManifestFiles);
+    }

```

Comment:
Shouldn't this logic be in `deployManifests` function?

Reply:
im routing only after deployments have rolled out successfully